src/main/java/org/minions/devfund/marcos/Squarelotron.java

```diff
@@ -0,0 +1,202 @@
+package org.minions.devfund.marcos;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+/**
+ * It is in charge to represent a square matrix and it is operations.
+ */
+public class Squarelotron {
+
+    private int[][] squarematrix;
+    private int size;
+
+    /**
+     * Initializes an instance of {@link Squarelotron}.
+     *
+     * @param n square matrix dimension.
+     */
+    public Squarelotron(final int n) {
+        this.size = n;
+        squarematrix = new int[n][n];
+        fillMatrix();
+    }
+
+    /**
+     * Gets the Squarelotron matrix.
+     *
+     * @return square matrix.
+     */
+    public int[][] getSquarematrix() {
+        return Arrays.copyOf(squarematrix, size);
+    }
+
+    /**
+     * Sets the Squarelotron matrix.
+     *
+     * @param squarematrix square matrix.
+     */
+    public void setSquarematrix(final int[][] squarematrix) {
+        this.squarematrix = squarematrix.clone();
+    }
+
+    /**
+     * Performs the Up Side Down Flip operation.
+     *
+     * @param ring matrix ring number.
+     * @return {@link Squarelotron}.
+     */
+    public Squarelotron upsideDownFlip(final int ring) {
+        Squarelotron newSquarelotron = new Squarelotron(this.size);
+        newSquarelotron.setSquarematrix(swapRows(this.squarematrix));
+        newSquarelotron.setSquarematrix(replaceMaskValues(newSquarelotron.getSquarematrix(), this.squarematrix, ring));
+        return newSquarelotron;
+    }
+
+    /**
+     * Performs the Up Main Diagonal Flip operation.
+     *
+     * @param ring matrix ring value.
+     * @return {@link Squarelotron}.
+     */
+    public Squarelotron mainDiagonalFlip(final int ring) {
+        Squarelotron newSquarelotron = new Squarelotron(this.size);
+        newSquarelotron.setSquarematrix(transposeMatrix(this.squarematrix));
+        newSquarelotron.setSquarematrix(replaceMaskValues(newSquarelotron.getSquarematrix(), this.squarematrix, ring));
+        return newSquarelotron;
+    }
+
+    /**
+     * Rotates the square matrix elements 90 degrees according the number of turns parameter.
+     *
+     * @param numberOfTurns number of turns value.
+     */
+    public void rotateRight(final int numberOfTurns) {
+        if (numberOfTurns >= 0) {

```

Comment:
You can improve this.
Instead of rotating all the time 
you can calculate the actual turn
if the numberOfTurns is 4 nothing should happen
if the numberOfTurns is 5 only it should rotate 1 time.